icicle/include/icicle/fields/field.h

```diff
@@ -83,6 +84,27 @@ class Field
     return omega;
   }
 
+  static HOST_INLINE Field
+  hex_str2scalar(const std::string& in_str) // The input string should be in a hex format - 0xABCD....
+  {
+    assert(in_str.substr(0, 2) == "0x" && "The input string is not in hex format!");
```

Comment:
note that assert is only in debug mode so consider using ICICLE_ASSERT

Reply:
No.
I didnt touch it